public/app/plugins/datasource/testdata/components/RawFrameEditor.tsx

```diff
@@ -1,21 +1,56 @@
-import React from 'react';
-import { InlineField, TextArea } from '@grafana/ui';
+import React, { useState } from 'react';
+import { Alert, CodeEditor } from '@grafana/ui';
 import { EditorProps } from '../QueryEditor';
+import { isArray } from 'lodash';
+import { toDataQueryResponse } from '@grafana/runtime';
+import { dataFrameToJSON } from '@grafana/data';
 
 export const RawFrameEditor = ({ onChange, query }: EditorProps) => {
-  const onContent = (rawFrameContent: string) => {
-    onChange({ ...query, rawFrameContent });
+  const [error, setError] = useState<string>();
+  const [warning, setWarning] = useState<string>();
+
+  const onSaveFrames = (rawFrameContent: string) => {
+    try {
+      const json = JSON.parse(rawFrameContent);
+      if (isArray(json)) {
+        setError(undefined);
+        setWarning(undefined);
+        onChange({ ...query, rawFrameContent });
+        return;
+      }
+
+      // Chek if it is a copy of the raw resuls
+      const v = toDataQueryResponse({ data: json });
+      if (v.data?.length && !v.error) {
+        const data = v.data.map((f) => dataFrameToJSON(f));
+        console.log('SOURCE', json);
```

Comment:
LGTM. 

nit: Queries work fine when you copy from the network response. But I think we should also handle "copy from Query inspector" scenario. ( Query Inspector -> JSON -> Data -> Copy and paste to see the `Unable to read dataframes in text` error )

nit: Is the logging intentional here?